bioconda_utils/artifacts.py

```diff
@@ -110,17 +116,24 @@ def upload_pr_artifacts(config, repo, git_sha, dryrun=False, mulled_upload_targe
     backoff.expo,
     requests.exceptions.RequestException
 )
-def download_artifact(url, to_path):
+def download_artifact(url, to_path, artifact_source):
     logger.info(f"Downloading artifact {url}.")
-    resp = requests.get(url, stream=True, allow_redirects=True)
+    headers = {}
+    if artifact_source == "github-actions":
+        token = os.environ.get("GITHUB_TOKEN")
+        if not token:
+            logger.critical("GITHUB_TOKEN required to download GitHub Actions artifacts")
+            exit(1)
+        headers = {"Authorization": f"token {token}"}
+    resp = requests.get(url, stream=True, allow_redirects=True, headers=headers)
```

Comment:
is there a way of guarding against possible leakage of token here?

Reply:
Any suggestions? PyGithub only provides the download URL, it doesn't have a method for downloading the artifact. I can make a request through that (though it's kinda hacky `artifact._requester`) which returns the temporary signed download URL and then download that.

Reply:
Another option could be to do something with https://github.com/actions/download-artifact

It's frustrating that GHA requires some sort of credentials to get artifacts.

Reply:
Hmm, I feel like the temporary signed download URL would be a more secure option. But IIRC GitHub will obfuscate secrets in logs even if inadvertently leaked (like a malicious PR with `print(token)`), and looking more closely, the token is used in a similar manner elsewhere in the code. In other words, this is not changing the current security stance, so I think it's fine.

Reply:
Thanks, @daler I think you are right about the secrets, as the docs just say to make sure to declare any that have been modified (e.g. base64 encoded) later on, etc.

For clarity/posterity, the GH API generates and redirects to that temporary signed URL from the download URL I'm using here. So, there's no way around sending the auth, it's just that using PyGitHub is already in use at this point in the code so if it was used there would be fewer places where we have to pass the token in Bioconda code. I think it's ultimately doing the same thing I'm doing here in sending a request to the API, just a bit more buried.